### PR TITLE
refactor(zentao): add custom types OperatedBy and StringFloat64 to simplify codes

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. The progress field in response of zentao's api is a string or a float, at #6089  we add a progressRes field to receive this field and convert it into progress(float64). 
2. So do `ClosedBy` and `CanceledBy` fields.

### Does this close any open issues?
Closes None

### Screenshots
Include any relevant screenshots here.
The main branch cannot running, so test is on the way.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Comment by d4x1** [comment]
Here is an example for `closedBy` field:
```json
{
    "closedBy": {
        "id": 1,
	"account": "admin",
	"avatar": "",
	"realname": "admin"
    },
    "others...": ""
}
```

File: backend/core/models/common/string_float64.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cast"
+)
+
+type StringFloat64 struct {
+	v float64
+	t string
```
Comment (klesh) [inline backend/core/models/common/string_float64.go:29]: I did the same for the `iso8061time`, which was never really useful. 
Should we ignore the original type to make it simpler?

Comment (d4x1) [reply backend/core/models/common/string_float64.go:29]: The original type is useless if we just want codes to run correctly, but if we define a new custom type, the original type is the basic component. I tend to keep type info in order to make the custom `StringFloat64` "unbroken".

Comment (klesh) [reply backend/core/models/common/string_float64.go:29]: Yup, I totally understand, that is why I did it for the `iso8601time`, like I said, it has never been used.
Seems like the `type` is only used in the  `Marshal` method which is actually not being called anywhere.

But, I'm OK with it for now.

**Review by klesh — APPROVED** [review]

**Pull request merged**